.github/actions/test-unit/action.yml

```diff
@@ -0,0 +1,49 @@
+name: 'Run unit tests'
+description: 'Check if a string is a number'
+inputs:
+  sinceMaster:
+    description: 'Whether to run tests since master'
+    required: false
+    type: boolean
+    default: false
+  codecov-token:
+    description: 'The environment name'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Cache Jest cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          .jest-cache/
+        key: jest-cache-${{ github.sha }}
+        restore-keys: |
+          jest-cache-
+    - name: Get number of CPU cores
+      id: cpu-cores
+      uses: SimenB/github-actions-cpu-cores@v1
+    - name: Fetch master branch
+      shell: bash
+      if: ${{ inputs.sinceMaster == 'true' }}
+      run: |
+        chown -R root /__w/asap-hub/asap-hub
+        git fetch --no-tags --depth=1 origin master
```

Comment:
is there a way we can use the Checkout Action?

Reply:
i tried but it didn't work for me